src/AppBundle/Twig/AppExtension.php

```diff
@@ -50,4 +80,29 @@ public function pronoun($gender)
 
         return 'hen';
     }
+
+    /**
+     * @param bool $wantToLearn
+     * @return string
+     */
+    public function wantToLearnName($wantToLearn)
+    {
+        return $wantToLearn? $this->translator->trans('New'): $this->translator->trans('Established');
+    }
+
+    /**
+     * @param User $user
+     * @return string
+     */
+    public function userCategoryNameString(User $user)
+    {
+        $categoryNames  = array_values($user->getCategoryNames());
+        if ($categoryNames > 1) {
+            $lastCategory   = array_pop($categoryNames);
+            $categories = implode(', ', $categoryNames) .' '.  $this->translator->trans('and') .' '. $lastCategory;
```

Comment:
Looks like this prints the last category twice. Both with the implode and from lastCategory.